components/omega/doc/design/Broadcast.md

```diff
@@ -0,0 +1,132 @@
+
+## 4 Design
+
+The OMEGA broadcast is mostly just wrappers around the MPI broadcast
+function with simplified arguments.
+
+### 4.1 Data types and parameters
+
+No new data types are needed.
+
+### 4.2 Methods
+
+We will have four forms of the Broadcast function, each returning an
+integer error code.  Note that in the interfaces below, we could
+template based on data type, but prefer to overload to keep the
+interfaces cleaner.
+
+#### 4.2.1 Broadcast from master task in default environment
+
+The first form is the simplest for a broadcast within the default 
+environment and from the master task:
+
+    int Broadcast([data type] value);
+
+where `[data type]` is one of the supported types (I4, I8 R4, R8, Real,
+boolean, std::string). In actual use, this would look like:
+
+    ierr = Broadcast(myIntVar);
+    ierr = Broadcast(myRealVar);
+    [etc for all data types]
+
+On the master task, the value would be broadcast and remain unchanged.
+The remaining tasks would receive the broadcast and store the value.
+
+#### 4.2.2 Broadcast from another rank within default environment
+
+This is similar to the above, but adds the additional argument
+for the source rank to broadcast from.
+
+    int Broadcast([data type] value,  ///< [in] value to be broadcast
+                  const int srcRank   ///< [in] rank to broadcast from
+                  );      // 
+
+#### 4.2.3 Broadcast from master rank within a different environment
+
+As in 4.2.1, but adds the machine environment as an argument:
+
+    int Broadcast([data type] value,    ///< [in] value to be broadcast
+                  const MachEnv subEnv, ///< [in] defined OMEGA environment
+                  );
+
+#### 4.2.4 Broadcast from another rank within a different environment
+
+As in 4.2.2, but adds the machine environment as an argument:
+
+    int Broadcast([data type] value,     ///< [in] value to be broadcast
+                  const MachEnv subEnv,  ///< [in] defined OMEGA environment
+                  const int srcRank      ///< [in] rank to broadcast from
+                  );
+
+#### 4.2.5 Broadcast of vector variables
+
+There will be interfaces identical to the above with the value argument 
+replaced by `std::vector<type> value` to broadcast a vector of values.
+
+## 5 Verification and Testing
+
+### 5.1 Test scalars of all data types
+
+A multi-processor test driver will initialize variables of all supported
+types to zero or empty values. The master rank will then set a non-zero
+value and broadcast to the remaining ranks. Each rank will verify the
+new value is the expected one.  
+  - tests requirement 2.1
+
+### 5.2 Test vectors of all types
+
+Repeat the above with vectors of all types using a set of non-zero
+values. 
+  - tests requirement 2.2
+
+### 5.3 Test broadcast from other ranks
+
+Repeat the above two tests except broadcasting from a rank other
+than the master rank.
+  - tests requirement 2.3
+
+### 5.4 Test within a subset environment
```

Comment:
In MPI parlance, does this refer to using an intergroup communicator?

Reply:
Not in this case - for these functions, they are only broadcasting within the given communicator. So if we created a subset group the call would only broadcast within that group's communicator. There will be a later functionality that will do general communication both within and between groups to rearrange data.